test/test_coretasks.py

```diff
@@ -156,6 +156,30 @@ def test_handle_rpl_channelmodeis(mockbot, ircfactory):
     assert mockbot.channels["#test"].modes["t"]
 
 
+def test_handle_rpl_channelmodeis_clear(mockbot, ircfactory):
+    """Test RPL_CHANNELMODEIS events clearing previous modes"""
+    rpl_channelmodeis = [
+        ":mercury.libera.chat",
+        "324",
+        "TestName",
+        "#test",
+        "+knlt",
+        "hunter2",
+        ":1",
+    ]
+    irc = ircfactory(mockbot)
+    irc.bot._isupport = isupport.ISupport(chanmodes=("b", "k", "l", "mnt", tuple()))
+    irc.channel_joined("#test", ["Alex", "Bob", "Cheryl"])
+    mockbot.on_message(" ".join(rpl_channelmodeis))
+    rpl_channelmodeis[4] = "+kl"
+    mockbot.on_message(" ".join(rpl_channelmodeis))
```

Comment:
I don't like using smart code like `rpl_channelmodeis[4] = "+kl"` because now I've to go back to the list, and check which one is the 4th index (which I'm bad at). I'd prefer to use a template string, like so:

```suggestion
    irc = ircfactory(mockbot)
    irc.bot._isupport = isupport.ISupport(chanmodes=("b", "k", "l", "mnt", tuple()))
    irc.channel_joined("#test", ["Alex", "Bob", "Cheryl"])

    rpl_channelmodeis = ":mercury.libera.chat 324 TestName #test {mode} hunter2 :1"
    mockbot.on_message(rpl_channelmodeis.format(mode='+knlt'))
    mockbot.on_message(rpl_channelmodeis.format(mode='+kl'))
```

This version highlight the `mode` part of the message, providing more intent.

While I'm at it, I prefer to have the IRC setup before the exact part that is tested here (the RPL_CHANNELMODEIS), for readability.

Reply:
Fair points, though I'll just split out all the mode bits.